File: Java/recursion/functional/Functional.java
package Java.recursion.functional;

import java.util.Scanner;

public class Functional {
    public static void main(String args[]){
        int n;
        Scanner scanner=new Scanner(System.in);
        System.out.println("Enter the numbe n1:");
        n=scanner.nextInt();

        int sum=sumOfNnumber(n);
        System.out.println("sum is:"+sum);
    }

    public static int sumOfNnumber(int n){
        if(n==0){
            return 0;
        }else{
            return n+sumOfNnumber(n-1);
        }
    }
}

File: CPP/recursion/functional/Functional.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

int sumOfNnumbers(int n){
    if(n==0){
        return 0;
    }else{
         return n+sumOfNnumbers(n-1);
    }
}

int main(){
    /*find the sum of first n number using functioanl recursion
    Functional recursion:-it return back some value*/
    int n;
    cout<<"Enter the number n:"<<endl;
    cin>>n;
    int sum=sumOfNnumbers(n);
    cout<<"sum of first "<<n<<" number is:"<<sum<<endl;
}